### ISSUE TITLE
[Test-Proxy] Preserve original format when returning recorded response body content

### ISSUE THREAD

**Issue by mccoyp** [Azure/azure-sdk-tools#5968]
## Context

Python's `azure-containerregistry` library has tests that read a JSON manifest file and send this content to the service. The service creates a digest based on the stream content that was sent; this content is whitespace-dependent. Therefore, the digest we get from e.g. `manifest_spaces.json`:
```json
{"foo": 2, "bar": "baz"}
```
will not match the digest we get from e.g. `manifest_no_spaces.json`:
```json
{"foo":2,"bar":"baz"}
```
This is because the former digest is calculated with additional spaces in the stream.

In live mode, tests that validate the locally-calculated digest with the service-returned digest succeed when using `manifest_spaces.json`, since spaces are used in both digest calculations. The problem is that these tests fail in playback -- we calculate the local digest using spaces, but the recorded service response is returned without spaces by the test proxy.

For example, here's a recorded manifest from a response body ([source](https://github.com/Azure/azure-sdk-assets/blob/fd8926fc58dfafad9975d9aacab54fcae46f3440/python/sdk/containerregistry/azure-containerregistry/tests/recordings/test_container_registry_client.pyTestContainerRegistryClienttest_set_docker_manifest_json.json#L1590-L1606) from the assets repository):
```json
"ResponseBody": {
  "config": {
    "digest": "sha256:feb5d9fea6a5e9606aa995e879d862b825965ba48de054caab5ef356dc6b3412",
    "mediaType": "application/vnd.docker.container.image.v1\u002Bjson",
    "size": 1469
  },
  "layers": [
    {
      "digest": "sha256:2db29710123e3e53a794f2694094b9b4338aa9ee5c40b930cb8063a1be392c54",
      "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
      "size": 2479
    }
  ],
  "mediaType": "application/vnd.docker.distribution.manifest.v2\u002Bjson",
  "schemaVersion": 2
}
```
When the test proxy sends the response body content in playback, the recorded JSON is written back without any whitespace -- no spaces between keys and values. Because of this, the digest that's calculated from the response body differs from the one calculated from the local manifest file.

## Feature request

If we had a way to request a certain formatting of the response body, for example with a regular expression, we could preserve the _actual_ response body format in playback. As things stand, the response body content in live mode doesn't exactly match the response body content in playback.

@scbedd I could probably help with this, but I would need assistance in figuring out where exactly the response body is fetched from the recording and sent back to the test.

**Comment by scbedd** [comment]
To _properly_ do this, we need to maintain _exactly_ the body that was sent. This will mean storing the Request/Response body **as base64 encoded value.**

We can either:

1. Add a new attribute, store the body there. Keep existing RequestBody for visibility only.
2. Replace RequestBody/ResponseBody w/ the base64 value.

In either case, we will need to update at the very least:

- [x] [`RecordEntry.DeserializeBody()`](https://github.com/Azure/azure-sdk-tools/blob/5257cee7373acc9b7782b1a93a2bc29a43e5f12c/tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/RecordEntry.cs#L85)
- [x] [`RecordEntry.SerializeBody()`](https://github.com/Azure/azure-sdk-tools/blob/5257cee7373acc9b7782b1a93a2bc29a43e5f12c/tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/RecordEntry.cs#L191)
- [x]  Update where we _return_ the selected value. [In RecordingHandler](https://github.com/Azure/azure-sdk-tools/blob/7620a2d95ce19f8f52fe4da6aef5afc25742dead/tools/test-proxy/Azure.Sdk.Tools.TestProxy/RecordingHandler.cs#L505).
- [x] Handle temp deserialization for [sanitization.](https://github.com/Azure/azure-sdk-tools/blob/5257cee7373acc9b7782b1a93a2bc29a43e5f12c/tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/RecordedTestSanitizer.cs#L55)

And any attachment points related to these.

**Comment by scbedd** [comment]
@mccoyp this is now resolved. You will need to re-record the tests, but we now treat these as non-text bodies so they are stored byte-for-byte as they come in.

Resolved in #8430 #8325
